akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ClusterShardingQuery.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.sharding.typed
+
+import akka.actor.typed.ActorRef
+import akka.cluster.sharding.ShardRegion.CurrentShardRegionState
+import akka.cluster.sharding.typed.scaladsl.EntityTypeKey
+
+/**
+ * Protocol for querying sharding state e.g. A ShardRegion's state
+ */
+sealed trait ClusterShardingQuery
+
+/**
+ * Query the ShardRegion state for the given entity type key. This will get the state of the
+ * local ShardRegion's state.
+ */
+final case class GetShardRegionState(entityTypeKey: EntityTypeKey[_], replyTo: ActorRef[CurrentShardRegionState]) extends ClusterShardingQuery {
+  /**
+   * Java API
```

Comment:
Perhaps we should define separate classes in scaladsl and javadsl? They would be very similar. Perhaps ok to share since those queries are not core functionality?

Reply:
`CurrentShardRegionState` and `ShardState` are currently shared (the responses) so perhaps leave this as shared for now.